src/main/java/com/gachigang/ontherun/service/CustomLogoutHandler.java

```diff
@@ -1,18 +1,43 @@
 package com.gachigang.ontherun.service;
 
+import com.gachigang.ontherun.persistence.entity.Token;
+import com.gachigang.ontherun.persistence.repository.TokenRepository;
 import jakarta.servlet.http.HttpServletRequest;
 import jakarta.servlet.http.HttpServletResponse;
-import lombok.AllArgsConstructor;
+import lombok.RequiredArgsConstructor;
+import org.springframework.lang.NonNull;
 import org.springframework.security.core.Authentication;
 import org.springframework.security.core.context.SecurityContextHolder;
 import org.springframework.security.web.authentication.logout.LogoutHandler;
 import org.springframework.stereotype.Service;
 
+import static com.gachigang.ontherun.common.ApplicationConstants.Security.TOKEN_PREFIX;
+
 @Service
-@AllArgsConstructor
+@RequiredArgsConstructor
 public class CustomLogoutHandler implements LogoutHandler {
+
+    private final TokenRepository tokenRepository;
+
     @Override
-    public void logout(HttpServletRequest request, HttpServletResponse response, Authentication authentication) {
-        SecurityContextHolder.clearContext();
+    public void logout(
+            @NonNull final HttpServletRequest request,
+            @NonNull final HttpServletResponse response,
+            @NonNull final Authentication authentication
+    ) {
+        final String authHeader = request.getHeader("Authorization");
+        final String jwt;
+        if (authHeader == null || !authHeader.startsWith(TOKEN_PREFIX)) {
+            return;
+        }
+        jwt = authHeader.substring(7);
+        Token storedToken = tokenRepository.findByToken(jwt)
+                .orElse(null);
```

Comment:
Seems strange, we need to use cool features of Optional instead of unwrapping it ... lets try to use `ifPresent()` istead of returning null if object null and checking on it